### PR TITLE
Unit Tests

### PR DESCRIPTION
**Pull request by cccs-kevin**
https://cccs.atlassian.net/browse/AL-639

To use, ensure you have pytest enabled on your PyCharm (https://www.jetbrains.com/help/pycharm/pytest.html) and installed in your Python Interpreter. Then right click on `test` directory and select "Run 'pytest in test' with Coverage". 

Things to do still:
- Finish writing unit tests for files in the configextractor service such as `configextractor.py`, `cli.py`, `wrapper_malconf.py`
- achieve 100% code coverage (currently at 92% in `configextractor.py` and 83% in `cli.py`)
- address TODOs
- refactor test_service file according to legit anyone else's opinions

### CODE REVIEW

File: configextractor.py
Comment (cccs-aa) [inline configextractor.py:123]: yes it should only ever be in a single parser_list

Comment (cccs-aa) [reply configextractor.py:123]: because it doesn't make any sense to have a parser that ran in multiple entries in the yara_parser.yaml

File: configextractor.py
Comment (cccs-aa) [inline configextractor.py:172]: fields would usually be smaller because it's the output from the parser

File: configextractor.py
Comment (cccs-aa) [inline configextractor.py:179]: it's either a list or a list of lists

File: configextractor.py
Comment (cccs-kevin) [reply configextractor.py:123]: Okay great. Next question... after running debug, if looks like `obj.parser_list` is always a list with a single item which is equal to `name`, and `parser` is always equal to one of the keys in `self.file_parsers`, so why are we even iterating at all? Why are we checking the `obj.parser_list` attribute when the value contained in it is the same as `name`?  Why not just do `obj = self.file_parsers[parser]` ?

File: configextractor.py
Comment (cccs-kevin) [reply configextractor.py:172]: Okay then I suggest something along the lines of:
```
for field_key, field_value in fields.items():
    if field_key in FIELD_TAG_MAP:
        tag = FIELD_TAG_MAP[field_key]
        ...
```

File: configextractor.py
Comment (cccs-kevin) [reply configextractor.py:179]: Okay cool, and I'm assuming if it's a list of lists then we want the nested lists to be tagged per nested list and not per item in the nested list ex. `"network.static.uri": (a, b, c) versus "network.static.uri": a, "network.static.uri": b, "network.static.uri": c`? Also we should add a check here that checks if the`tag` exists in `table_section` already, and then not try to add it if it does.

File: configextractor.py
Comment (cccs-aa) [reply configextractor.py:123]: not quite, parser_list is usually just a single item but it can be more. This is more for future reasons if you need to run multiple parsers on one yara rule hit. It's not really being used currently. parser is actually not always equal to one of the keys in self.file_parsers. There's two different names here. parser will always be one of the fields under the parser - MWCP block in the yaml. The keys in self.file_parsers are the names in the start of each block in yara_parser.yaml see here 
https://github.com/CybercentreCanada/assemblyline-service-configextractor/blob/master/yara_parser.yaml
Now they look the same here but in the proprietary version there are differences and this handles that. That's why it's like this.

File: configextractor.py
Comment (cccs-aa) [reply configextractor.py:172]: done

Comment (cccs-kevin) [reply configextractor.py:172]: where?

File: configextractor.py
Comment (cccs-kevin) [reply configextractor.py:123]: okay, then if that is the case, then I suggest that after we find a hit for `if parser in obj.parser_list` and we assign all of the variables, we break out of the `for` loop with `break` since there is no point of continuing to iterate through `self.file_parsers.items()` looking for hits given that there should only be one hit ever.

File: configextractor.py
Comment (cccs-aa) [reply configextractor.py:179]: so right now we aren't tagging anything that's a listofstringtuples. all the values in FIELD_TAG_MAP are listofstrings. Also sometimes it doesn't really make much sense to tag them because values in listofstringtuples are related but different and so don't really belong under one tag. For example  
https://github.com/Defense-Cyber-Crime-Center/DC3-MWCP/blob/0d27dea113751ef1aff727aaa2cf13c709e6beab/mwcp/config/fields.json#L252
As well AL already does the deduplication for tags. Would it improve readability if we explicitly add a check?

File: configextractor.py
Comment (cccs-aa) [reply configextractor.py:171]: just pushed it now 
https://github.com/CybercentreCanada/assemblyline-service-configextractor/commit/18c0690310a356368cf839ae113c310971d4849d

File: configextractor.py
Comment (cccs-aa) [reply configextractor.py]: done here
https://github.com/CybercentreCanada/assemblyline-service-configextractor/commit/4fedc0be87b62d6395a865bbf606daf5f839076a

File: configextractor.py
Comment (cccs-kevin) [reply configextractor.py:179]: Okay good to know.

Valid point.

Fair, us explicitly checking if the tag exists would be the same efficiency as calling add_tag from what I can tell, so we should be good with status quo. 

Resolving convo.

**Review by cccs-kevin — COMMENTED** [review]
💪

**Review by cccs-jp — APPROVED** [review]

File: test/test_service.py
Comment (cccs-kevin) [inline test/test_service.py:28]: either we call this filename "sample_txt" or we change filename='mirai_strings' -> filename=<sha-256> of file. your call

Comment (cccs-aa) [reply test/test_service.py:28]: changed mirai_strings to sha256

File: test/test_service.py
Comment (cccs-kevin) [inline test/test_service.py:14]: We should create a list named `samples`, containing `sample1` and `sample2`

File: test/test_service.py
Comment (cccs-kevin) [inline test/test_service.py:317]: We should pass the `samples` list here, rather than `[sample1]`

File: test/test_service.py
Comment (cccs-kevin) [inline test/test_service.py:327]: This should be `service_task = ServiceTask(sample)`

**Review by cccs-kevin — COMMENTED** [review]
I left some comments regarding using the `sample` dicts in the `test_execute` method. Once that is addressed and the conflicts are resolved, this PR has my approval 🎖️

**Pull request merged**
